### ISSUE TITLE
sh: mpv: command not found

### ISSUE THREAD

**Issue by frozenpandaman** [ekisu/mpv-webm#135]
## mpv version and platform

macOS 12.0.1
```
$ mpv --version                                                  
mpv 0.33.1 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on Tue Nov  2 06:21:53 UTC 2021
FFmpeg library versions:
   libavutil       56.70.100
   libavcodec      58.134.100
   libavformat     58.76.100
   libswscale      5.9.100
   libavfilter     7.110.100
   libswresample   3.9.100
FFmpeg version: 4.4.1
```

## Description

Installing this fresh on a new machine, all encodes are failing because for some reason the script can't find `mpv`, even though it's installed on the command line. The proper `run_subprocess_popen: running mpv ...` command is triggering and successfully being logged, and manually copy-pasting that command into new Terminal session works, but I'm getting the following right after it when it's being run via the script in mpv:
```
[v][webm] "sh: mpv: command not found" 
```

But if I launch sh (`/bin/sh`) it knows what the `mpv` command is, which I installed via Homebrew.

Hopefully I'm just overlooking something simple here! Thanks.

**Comment by frozenpandaman** [comment]
>manually copy-pasting that command into a new Terminal session works

I've discovered this the above is true when the output format is set to webm, though if converting to mp4 (h264/AAC) I get:
```
[encode] codec 'libx264' not found.
 (+) Video --vid=1 (*) (h264 320x240 15.000fps)
 (+) Audio --aid=1 (*) (aac 6ch 48000Hz)
[encode] codec 'libx264' not found.
Error opening/initializing the selected video_out (--vo) device.
Video: no video

[encode] no data written to target file
```

But I do have it installed:
```
$ x264 --version                                                 
x264 0.163.3060 5db6aa6
built on Oct 22 2021, clang: 13.0.0 (clang-1300.0.29.3)
x264 configuration: --chroma-format=all
libx264 configuration: --chroma-format=all
x264 license: GPL version 2 or later
```

**Comment by ekisu** [comment]
Does setting `display_progress = false` on the options work? When using `display_progress` we use Lua's `io.popen` function, it might have some different interactions with how macOS/Homebrew works.

As for the libx264 codec, it probably has something with the ffmpeg library linked with this mpv build. Running `mpv --ovc=help` should list the supported codecs

**Comment by frozenpandaman** [comment]
>Does setting `display_progress = false` on the options work?

Nope, tried putting this in script-opts/webm.conf and still says encode failed with the log message `sh: mpv: command not found`. I tried changing default shells, etc. and it doesn't seem to affect this – for some reason from within mpv.app it doesn't know where the `mpv` binary is or what that command is. If there's a way to make it use `ffmpeg` instead, I could maybe try that (and see if it also can't find that command – but as this script's not supposed to have external dependencies I guess that's not ideal either).

>Running `mpv --ovc=help` should list the supported codecs

libx264 is indeed in that list:
```
Available output video codecs:
  [...]
  --ovc=libx264      libx264 H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
  [...]
```
Full output [here](https://www.toptal.com/developers/hastebin/evasoravox) if you're interested. But I guess this is a secondary issue (as this is only when I paste one command into Terminal manually)… not even getting this within mpv's log file yet or anything, it's always still "mpv: command not found".

**Comment by frozenpandaman** [comment]
>I tried changing default shells, etc. and it doesn't seem to affect this

Turns out I was wrong about this and didn't test it correctly! Tried making `/bin/sh` my default shell, and I get (as expected!)
```
-sh: mpv: command not found
```

I think this is an M1 Mac/ARM64 problem – because Homebrew installs into `/usr/local` on Intel Macs (which is in my `sh` PATH) but installs into `/opt/homebrew` on Apple Silicon (which is not on my PATH), i.e. `mpv` used to get installed by brew into somewhere that was default to the `sh` path previously, and now it does not.

I can manually add Homebrew to my `sh` PATH by adding `export PATH="$PATH:/opt/homebrew/bin"` to the `~/.profile` file – but I think a better solution would be to modify the script's `run_subprocess()` function to use whatever your default shell is, not `sh`.

**Comment by sherlock-png** [comment]
Hello, my computer is M4 Mac, i have the same queastion. I download the mpv use homebrew, when use the webm cut video failed, the log mpv.txt says [   3.950][v][webm] "sh: mpv: command not found". 

Do you have a way to solve this problem,thank you very much, i really need this tool!!!

**Comment by sherlock-png** [comment]
@frozenpandaman Thank you very much, i have a seem question,through your function,this question can be solved!!!

1 symlink the mpv binary into /usr/local/bin
```
$ sudo ln -s /opt/homebrew/bin/mpv /usr/local/bin/ 
```

2 $ mpv --ovc=help
cant find the libx264,so the output file only audio
we need to change the file webm.lua
about 1050 line
change -- self.videoCodec = "libx264" to self.videoCodec = "h264_videotoolbox"

**Comment by sherlock-png** [comment]
@frozenpandaman  Thank you very much, i have a seem question,through your function,this question can be solved!!!

1 symlink the mpv binary into /usr/local/bin
```
$ sudo ln -s /opt/homebrew/bin/mpv /usr/local/bin/ 
```

2 change the encoder
$ mpv --ovc=help
cant find the libx264,so the output file only audio
we need to change the file webm.lua
about 1050 line
change -- self.videoCodec = "libx264" to self.videoCodec = "h264_videotoolbox"

sorry my english is bad, and i only can conclusion good person's idea, i really thanks @frozenpandaman. you are the best!
in chinese,we ask someone like you use "大佬"
